sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -293,9 +295,10 @@ class DataFrame protected[sql](
    *   1983  03    0.410516        0.442194
    *   1984  04    0.450090        0.483521
    * }}}
+   * @param numRows Number of rows to show
    * @group basic
    */
-  def show(): Unit = println(showString())
+  def show(numRows: Int = 20): Unit = println(showString(numRows))
```

Comment:
can you try df.show()?

in either case, I wouldn't not want to rely on Scala's generated code for default param values for bytecode level binary compatibility. Please implement two show methods, one with an int param and one without. Thanks.